model/indices/data_stream.smithy

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+structure DataStream{
+    name:String,
+    timestamp_field:AttributesMap,
+    indices: IndicesList
+    generation: Integer,
+    status: String,
+    template: String
+}
+
+list IndicesList{
+    member: AttributesMap
```

Comment:
Even IF `timestamp_field` and `indices` have the exact same schema, we should still define two different models instead of using a generic one called `AttributesMap`. More over, from what I read in https://opensearch.org/docs/latest/im-plugin/data-streams/, `indices` have a more defined structure than Attributes map.

This should be
```
structure DataStream{
    name: String,  # Remember to put space after the colon as well.
    timestamp_field: DataStreamTimestampField,
    indices: DataStreamIndices
    generation: Integer,
    status: String,
    template: String
}

list DataStreamIndices {
    member: DataStreamIndex
}

structure DataStreamIndex {
  index_name: String,
  index_uuid: String
}

map DataStreamTimestampField {
  key: String
  value: String
}
```

Reply:
Oh, i got it. Thank you! You can review it again